### PR TITLE
Allow passing DNS servers and search prefixes to build container

### PR DESCRIPTION
**Pull request by csrwng**
Allows origin code to pass in DNS lookup info to container launched by STI

### CODE REVIEW

**Comment by csrwng** [comment]
@mfojtik @soltysh ptal
This is the first part of the issue here: https://bugzilla.redhat.com/show_bug.cgi?id=1225410
@danmcp fyi

**Comment by mfojtik** [comment]
LGTM

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
[merge]

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -346,9 +348,16 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 	if opts.Stdout != nil {
 		config.AttachStdout = true
 	}
+	var hostConfig *docker.HostConfig
+	if opts.DNS != nil || opts.DNSSearch != nil {
```
Comment (bparees) [inline pkg/docker/docker.go]: what's the docker behavior if only DNSSearch but not DNS is set here?  (ie DNS ends up getting set to nil)

**Comment by smarterclayton** [comment]
Receiving reports container DNS is broken after this change

**Comment by bparees** [comment]
has this change even been applied to origin?  i'm not seeing a bump in origin for STI since may 15th.

**Comment by soltysh** [comment]
Nope, I'm waiting for #207 and updating it.

**Comment by csrwng** [comment]
We don't need this change anymore since we're going the documentation route. Should I submit a PR to reverse it?

**Comment by soltysh** [comment]
Yeah, if it's not needed anymore please remove it. I'll merge it and I'll
update origin with latest version.

**Comment by rassie** [comment]
@csrwng: which change to the documentation do you refer to? I'm encountering https://github.com/openshift/origin/issues/2598 and this PR is supposed to be the solution.

**Comment by csrwng** [comment]
@rassie instead of having the s2i builder pass the nameserver and search entries to the container it launches, simply add the openshift master as a nameserver to the node's resolv.conf

**Comment by smarterclayton** [comment]
Unfortunately the search entries aren't passed along.  This means that namespace local searches don't work.  We probably need to "pass on" the correct DNS config (which may be different than the builder) to s2i run

**Pull request merged**